taichi/transforms/constant_fold.cpp

```diff
@@ -135,8 +65,85 @@ class ConstantFold : public BasicStmtVisitor {
       }
     }
 
-    if (jit_evaluate_binary_op(new_constant, stmt, lhs->val, rhs->val)) {
-      insert_and_erase(stmt, new_constant);
+    // Type check should have been done at this point.
+    auto dt = lhs->val.dt;
+    switch (stmt->op_type) {
+#define COMMA ,
+#define HANDLE_REAL_AND_INTEGRAL_BINARY(OP_TYPE, PREFIX, OP_CPP)             \
+  case BinaryOpType::OP_TYPE: {                                              \
+    if (dt->is_primitive(PrimitiveTypeID::f32) ||                            \
+        dt->is_primitive(PrimitiveTypeID::f64)) {                            \
+      auto res = TypedConstant(                                              \
+          dst_type, PREFIX(lhs->val.val_cast_to_float64()                    \
+                               OP_CPP rhs->val.val_cast_to_float64()));      \
+      insert_and_erase(stmt, res);                                           \
+    } else if (dt->is_primitive(PrimitiveTypeID::i32) ||                     \
+               dt->is_primitive(PrimitiveTypeID::i64)) {                     \
+      auto res = TypedConstant(                                              \
+          dst_type, PREFIX(lhs->val.val_int() OP_CPP rhs->val.val_int()));   \
+      insert_and_erase(stmt, res);                                           \
+    } else if (dt->is_primitive(PrimitiveTypeID::u32) ||                     \
+               dt->is_primitive(PrimitiveTypeID::u64)) {                     \
+      auto res = TypedConstant(                                              \
+          dst_type, PREFIX(lhs->val.val_uint() OP_CPP rhs->val.val_uint())); \
+      insert_and_erase(stmt, res);                                           \
+    }                                                                        \
+    break;                                                                   \
+  }
+
+      HANDLE_REAL_AND_INTEGRAL_BINARY(mul, , *)
+      HANDLE_REAL_AND_INTEGRAL_BINARY(add, , +)
+      HANDLE_REAL_AND_INTEGRAL_BINARY(sub, , -)
+      HANDLE_REAL_AND_INTEGRAL_BINARY(truediv, , /)
```

Comment:
`truediv` should all be transformed into `div` in https://github.com/taichi-dev/taichi/blob/6a0b63ef709ae4da836492712d7f43c2c226d479/taichi/transforms/type_check.cpp#L318-L329